component/common/appendable/appendable.go

```diff
@@ -69,25 +63,25 @@ func (app *flowAppender) Append(ref storage.SeriesRef, l labels.Labels, t int64,
 		set := make([]*metrics.FlowMetric, 0)
 		app.buffer[t] = set
 	}
-	// If ref is 0 then lets grab a global id
-	if ref == 0 {
-		ref = storage.SeriesRef(metrics.GlobalRefMapping.GetOrAddGlobalRefID(l))
-	}
-	// If it is stale then we can remove it
+	// The incoming ref is a global refid. The refid might be 0, and in that case assigning it to a flow metric
+	// will ensure that it is not 0.
+	fm := metrics.NewFlowMetric(metrics.RefID(ref), l, v)
+	// If it is stale then we can remove it.
 	if value.IsStaleNaN(v) {
-		metrics.GlobalRefMapping.AddStaleMarker(uint64(ref), l)
+		metrics.GlobalRefMapping.AddStaleMarker(fm.GlobalRefID(), l)
 	} else {
-		metrics.GlobalRefMapping.RemoveStaleMarker(uint64(ref))
+		metrics.GlobalRefMapping.RemoveStaleMarker(fm.GlobalRefID())
 	}
-	app.buffer[t] = append(app.buffer[t], &metrics.FlowMetric{
-		GlobalRefID: uint64(ref),
-		Labels:      l,
-		Value:       v,
-	})
+	app.buffer[t] = append(app.buffer[t], fm)
 	return ref, nil
 }
 
-func (app *flowAppender) AppendExemplar(ref storage.SeriesRef, l labels.Labels, e exemplar.Exemplar) (storage.SeriesRef, error) {
+func (app *flowAppender) AppendExemplar(
+	ref storage.SeriesRef,
+	l labels.Labels,
+	e exemplar.Exemplar,
+) (storage.SeriesRef, error) {
+
 	return 0, nil
 }
```

Comment:
Can you add a TODO here? This is something we'll want to implement before release.